src/interpreter/mod.rs

```diff
@@ -169,14 +169,32 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
     // TODO(solson): Try making const_to_primval instead.
     fn const_to_ptr(&mut self, const_val: &const_val::ConstVal) -> EvalResult<'tcx, Pointer> {
         use rustc::middle::const_val::ConstVal::*;
+        use rustc_const_math::{ConstInt, ConstIsize, ConstUsize};
+        macro_rules! i2p {
+            ($i:ident, $n:expr) => {{
+                let ptr = self.memory.allocate($n);
+                self.memory.write_int(ptr, $i as i64, $n)?;
+                Ok(ptr)
+            }}
+        }
         match *const_val {
             Float(_f) => unimplemented!(),
-            Integral(int) => {
-                // TODO(solson): Check int constant type.
-                let ptr = self.memory.allocate(8);
-                self.memory.write_uint(ptr, int.to_u64_unchecked(), 8)?;
```

Comment:
Heh, it never occurred to me that this was only working for smaller integers because of little-endianness. Good fix.